tests/update/tests.py

```diff
@@ -199,3 +202,18 @@ def test_update_with_joined_field_annotation(self):
             with self.subTest(annotation=annotation):
                 with self.assertRaisesMessage(FieldError, msg):
                     RelatedPoint.objects.annotate(new_name=annotation).update(name=F('new_name'))
+
+
+@unittest.skipUnless(connection.vendor == 'mysql', 'MySQL tests')
+class MySQLUpdateTest(TestCase):
+    @classmethod
+    def setUpTestData(cls):
+        cls.u1 = UniqueNumber.objects.create(number=1)
+        cls.u2 = UniqueNumber.objects.create(number=2)
+
+    def test_order_by_update_on_unique_constraint(self):
+        """
+        Update a queryset using ordered_by on a unique constraint.
+        """
+        updated_count = UniqueNumber.objects.all().order_by('-number', 'id').update(number=F('number') + 1)
```

Comment:
The `.all()` is redundant
```suggestion
        updated_count = UniqueNumber.objects.order_by('-number', 'id').update(number=F('number') + 1)
```